src/Microsoft.AspNetCore.Mvc.RazorPages/Infrastructure/DefaultPageFactory.cs

```diff
@@ -57,10 +57,11 @@ public class DefaultPageFactory : IPageFactoryProvider
                 _modelMetadataProvider,
                 _propertyAccessors);
 
-            return (context) =>
+            return (PageContext context) =>
             {
                 var page = (Page)activatorFactory(context);
                 page.PageContext = context;
+                page.Path = context?.HttpContext?.Request.Path;
```

Comment:
You want https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.RazorPages/PageActionDescriptor.cs#L37